platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/AbstractLongList.java

```diff
@@ -180,7 +182,7 @@ protected AbstractLongList(final int numLongsPerChunk, final long maxLongs, fina
      * @param path File to read header from
      * @throws IOException If there was a problem reading the file
      */
-    protected AbstractLongList(final Path path, final long reservedBufferLength) throws IOException {
+    protected void init(final Path path, final long reservedBufferLength) throws IOException {
```

Comment:
What's the motivation for these method? You made all the fields non-final to make it work. There has to be a critically important reason. What is it?

Reply:
The idea was to have `tempFile = createTempFile(file.toFile().getName(), fileSystemManager)` for `LongListDisk` in its constructor and update only this concrete type's constructor to require `fileSystemManager` instead of having it in `AbstractLongList` constructor or as it was initially a side effect of calling the `LongListDisk` override of the  abstract `readBodyFromFileChannelOnInit` method.

Reply:
Fixed.

Reply:
Added [this commit](https://github.com/hashgraph/hedera-services/pull/14695/commits/41f0017db88dbfa93ea61a9345b83df1fee2342a) to make the fields final again and instead pass the `fileSystemManager` in `AbstractLongList` constructor.